atomicfu/src/nativeMain/kotlin/kotlinx/atomicfu/locks/PosixLockSupport.kt

```diff
@@ -0,0 +1,51 @@
+package kotlinx.atomicfu.locks
+
+import interop.*
+import kotlinx.cinterop.*
+import platform.posix.*
+import kotlin.concurrent.*
+
+public class NativeMutexNode {
+    private val mutex: CPointer<lock_support_t> = lock_support_init()!!
+
+    internal var next: NativeMutexNode? = null
+
+    fun lock() {
+        interop.lock(mutex)
+    }
+
+    fun unlock() {
+        interop.unlock(mutex)
+    }
+}
+
+/**
+ * This is a trivial counter-part of NativeMutexNode that does not rely on interop.def
+ * The problem is, commonizer cannot commonize pthreads, thus this declaration should be duplicated
+ * over multiple Native source-sets to work properly
+ */
+//public class NativeMutexNode {
```

Comment:
Do you expect commonizer being able to unlock this implementation one day? Do we really need to keep it here?

Reply:
This, or that we'll decide to give up on the commonizer and just copy-paste it over a few native sourcesets